<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Kaixuan Luo" />

<meta name="date" content="2023-12-05" />

<title>Detect LD mismatch between GWAS zscores and UKBB LD reference using DENTIST and SuSiE_RSS</title>

<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">Detect LD mismatch between GWAS zscores and UKBB LD reference using DENTIST and SuSiE_RSS</h1>
<h4 class="author">Kaixuan Luo</h4>
<h4 class="date">2023-12-05</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-12-05
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree8d658c82f8df82f45d7f483c6c99d23e7f8a219dtargetblank8d658c8a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/8d658c82f8df82f45d7f483c6c99d23e7f8a219d" target="_blank">8d658c8</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree8d658c82f8df82f45d7f483c6c99d23e7f8a219dtargetblank8d658c8a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/8d658c82f8df82f45d7f483c6c99d23e7f8a219d" target="_blank">8d658c8</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/detect_LD_mismatch_gwas_UKBBref_susie_rss.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd</code>) and HTML (<code>docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/8d658c82f8df82f45d7f483c6c99d23e7f8a219d/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">8d658c8</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-05
</td>
<td>
update plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/599682ee0804e629e1f8f104673c0c7bc098ed96/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">599682e</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
<td>
add some diagnosis plots
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/e0f339f3cba9c8b36e452a7c3f004ed877977a62/docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html" target="_blank">e0f339f</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/ed98001ded556cbbe8d297edc9d121f4bd53801b/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">ed98001</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
minor update on the messages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/cbdb9e4a39ece4c23bf5e6e46a0aa1c81da382db/docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html" target="_blank">cbdb9e4</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/93529bbfea7dcbc8224b176729c428bba3077d66/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">93529bb</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
minor update on the messages
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/af426aed4c2514c099944eef32a671502b986e89/docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html" target="_blank">af426ae</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/8883dec860626b3e5063b2d96919a988f18db805/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">8883dec</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
update susie_rss LDL result with chr6 result
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/c9fee4b2ec51db36c4baeef34082fc6673714018/docs/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.html" target="_blank">c9fee4b</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/c19525208b70f65e0523cbc716ae2836119e9c08/analysis/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd" target="_blank">c195252</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-11-29
</td>
<td>
compare DENTIST vs. susie_rss results
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages-and-functions" class="section level2">
<h2>Load packages and functions</h2>
<pre class="r"><code>library(ctwas)
library(susieR)
library(foreach)
library(data.table)
library(tidyverse)
suppressMessages(library(GenomicRanges))
suppressMessages(library(rtracklayer))</code></pre>
<pre class="r"><code># Load UKBB reference LD matrix and SNP info
load_UKBB_R_snp_info &lt;- function(region_df, ld_R_dir, filestem = &quot;ukb_b38_0.1&quot;){
  filename &lt;- sprintf(&quot;%s_chr%s.R_snp.%d_%d&quot;, filestem,
                      gsub(&quot;chr&quot;, &quot;&quot;, region_df$chr), region_df$start, region_df$stop)
  print(filename)
  if(!file.exists(file.path(ld_R_dir, paste0(filename, &quot;.RDS&quot;))) || !file.exists(file.path(ld_R_dir, paste0(filename, &quot;.Rvar&quot;)))){
    stop(&quot;LD Reference files not exist!&quot;)
  }
  R_snp &lt;- readRDS(file.path(ld_R_dir, paste0(filename, &quot;.RDS&quot;)))
  R_snp_info &lt;- read.table(file.path(ld_R_dir, paste0(filename, &quot;.Rvar&quot;)), header=TRUE)
  return(list(R_snp = R_snp, R_snp_info = R_snp_info))
}

# Match GWAS sumstats with LD reference files. Only keep variants included in LD reference.
match_gwas_R_snp &lt;- function(sumstats, R, snp_info){
  sumstats &lt;- sumstats[sumstats$id %in% snp_info$id,]
  R_snp_index &lt;- na.omit(match(sumstats$id, snp_info$id))
  sumstats$R_snp_index &lt;- R_snp_index
  R &lt;- R[R_snp_index, R_snp_index]
  stopifnot(nrow(sumstats) == nrow(R))
  return(list(sumstats = sumstats, R = R))
}

liftOver_hg19ToHg38 &lt;- function(gr){
  seqlevelsStyle(gr) &lt;- &quot;UCSC&quot;
  ch &lt;- import.chain(&quot;~/softwares/liftOver/hg19ToHg38.over.chain&quot;)
  gr &lt;- unlist(liftOver(gr, ch))
  genome(gr) &lt;- &quot;hg38&quot;
  return(gr)
}</code></pre>
</div>
<div id="ld-regions-ldetect-blocks" class="section level2">
<h2>LD Regions (ldetect blocks)</h2>
<pre class="r"><code>regions &lt;- system.file(&quot;extdata/ldetect&quot;, &quot;EUR.b38.bed&quot;, package = &quot;ctwas&quot;)
regions_df &lt;- read.table(regions, header = T)
regions_df &lt;- regions_df %&gt;% dplyr::arrange(chr, start, stop) %&gt;% dplyr::mutate(locus = 1:nrow(regions_df))</code></pre>
</div>
<div id="ldl" class="section level2">
<h2>LDL</h2>
<pre class="r"><code>trait &lt;- &quot;LDL&quot;</code></pre>
<pre class="r"><code>sumstats &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/gwas/gwas_processed/LDL-ukb-d-30780_irnt.sumstats.RDS&quot;)
# sumstats$z &lt;- sumstats$beta/sumstats$se
# z_snp &lt;- sumstats[,c(&quot;ID&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;z&quot;, &quot;N&quot;)]
# colnames(z_snp) &lt;- c(&quot;id&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;z&quot;, &quot;ss&quot;)
# z_snp &lt;- z_snp[!(z_snp$id %in% z_snp$id[duplicated(z_snp$id)]),] #drop multiallelic variants (id not unique)

sumstats &lt;- makeGRangesFromDataFrame(sumstats, start.field = &quot;pos&quot;, end.field = &quot;pos&quot;, keep.extra.columns = T)
sumstats_hg38 &lt;- liftOver_hg19ToHg38(sumstats) %&gt;% as.data.frame() %&gt;%
    dplyr::rename(chr = seqnames, pos = start, snp = SNP) %&gt;%
    dplyr::select(chr, pos, snp, A1, A2, beta, se, p, freq)
sumstats_hg38 &lt;- mapgen::assign_snp_locus(sumstats_hg38, regions_df)</code></pre>
<div id="compare-dentist-and-susie-rss-pvalues-in-chr22" class="section level3">
<h3>Compare DENTIST and SuSiE RSS pvalues in chr22</h3>
<p>load DENTIST results</p>
<pre class="r"><code>CHR=22
dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;LDL-ukb-d-30780_irnt.ukb_chr&quot;, CHR, &quot;.b38.DENTIST.full.txt&quot;))
dentist.chr.df &lt;- data.table::fread(dentist.res.file)
colnames(dentist.chr.df) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)

dentist.chr.snps &lt;- dentist.chr.df$rsID
dentist_detected_snps &lt;- dentist.chr.df$rsID[which(dentist.chr.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d variants with DENTIST result in chr%s \n&quot;, length(dentist.chr.snps), CHR))
cat(sprintf(&quot;%d detected variants with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist_detected_snps)))</code></pre>
<pre><code># 118796 variants with DENTIST result in chr22 
# 0 detected variants with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>Load Allele Frequency</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.chr.freq.df &lt;- data.table::fread(file.path(dentist.dir, paste0(&quot;LDL-ukb-d-30780_irnt.ukb_chr&quot;, CHR, &quot;.b38.frq&quot;)))</code></pre>
<p>load SuSiE RSS result</p>
<pre class="r"><code>select_loci &lt;- paste0(&quot;chr&quot;, CHR)
susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist_chr &lt;- readRDS(file.path(susie_rss_dir, paste0(trait, &quot;.condz.dist.&quot;, select_loci, &quot;loci.rds&quot;)))
condz_dist_chr.df &lt;- do.call(rbind.data.frame, condz_dist_chr)

susierss.chr.snps &lt;- condz_dist_chr.df$id
susierss_detected_snps &lt;- condz_dist_chr.df$id[which(condz_dist_chr.df$p_diff &lt; 5e-8)]
susierss_detected_flipped_snps &lt;- condz_dist_chr.df$id[which(condz_dist_chr.df$logLR &gt; 2 &amp; abs(condz_dist_chr.df$z) &gt; 2)]
cat(sprintf(&quot;%d variants with susie_rss result in chr%s \n&quot;, length(susierss.chr.snps), CHR))
cat(sprintf(&quot;%d detected variants with susie_rss pvalue &lt; 5e-8.\n&quot;, length(susierss_detected_snps)))
cat(sprintf(&quot;%d detected variants with susie_rss allele flipped.\n&quot;, length(susierss_detected_flipped_snps)))</code></pre>
<pre><code># 120845 variants with susie_rss result in chr22 
# 93 detected variants with susie_rss pvalue &lt; 5e-8.
# 3 detected variants with susie_rss allele flipped.</code></pre>
<div id="compare-susie-rss-with-dentist-result" class="section level4">
<h4>Compare SuSiE RSS with DENTIST result</h4>
<pre class="r"><code>cat(sprintf(&quot;%d variants with DENTIST result \n&quot;, length(dentist.chr.snps)))
cat(sprintf(&quot;%d variants with susie_rss result \n&quot;, length(susierss.chr.snps)))
cat(sprintf(&quot;%d variants with both DENTIST and susie_rss result \n&quot;, length(intersect(dentist.chr.snps, susierss.chr.snps))))
cat(sprintf(&quot;%d variants detected by both DENTIST and susie_rss (pvalue &lt; 5e-8).\n&quot;, length(intersect(dentist_detected_snps, susierss_detected_snps))))

common.snps &lt;- intersect(dentist.chr.snps, susierss.chr.snps)
df &lt;- data.frame(rsID = common.snps, dentist.pval = NA, susie.pval = NA)
m.dentist &lt;- match(common.snps, dentist.chr.df$rsID)
df$dentist.pval &lt;- dentist.chr.df$LP[m.dentist]
m.susie &lt;- match(common.snps, condz_dist_chr.df$id)
df$susie.pval &lt;- -log10(condz_dist_chr.df$p_diff[m.susie])
df$z &lt;- condz_dist_chr.df$z[m.susie]
df$condmean &lt;- condz_dist_chr.df$condmean[m.susie]
df$logLR &lt;- condz_dist_chr.df$logLR[m.susie]
df$Freq_A1 &lt;- dentist.chr.freq.df$Freq_A1[match(common.snps, dentist.chr.freq.df$RS_ID)]
df$MAF &lt;- pmin(df$Freq_A1, 1-df$Freq_A1)

m.sumstats &lt;- match(common.snps, sumstats_hg38$snp)
df$p &lt;- sumstats_hg38$p[m.sumstats]
df$locus &lt;- sumstats_hg38$locus[m.sumstats]
df$freq_A1_GWAS &lt;- sumstats_hg38$freq[m.sumstats]
df$MAF_GWAS &lt;- pmin(df$freq_A1_GWAS, 1-df$freq_A1_GWAS)

df &lt;- df %&gt;% dplyr::mutate(allele_flipping = ifelse(logLR &gt; 2 &amp; abs(z) &gt; 2, 1, 0))

df &lt;- df %&gt;% dplyr::mutate(type = case_when(susie.pval &gt; -log10(5e-8) &amp; dentist.pval &gt; -log10(5e-8) ~ &quot;SuSiE &amp; DENTIST&quot;, 
                                      susie.pval &gt; -log10(5e-8) ~ &quot;SuSiE only&quot;,
                                      dentist.pval &gt; -log10(5e-8) ~ &quot;DENTIST only&quot;,
                                      .default = &quot;null&quot;))

df$type &lt;- factor(df$type, levels = c(&quot;null&quot;,  &quot;SuSiE only&quot;, &quot;DENTIST only&quot;, &quot;SuSiE &amp; DENTIST&quot;))
table(df$type)</code></pre>
<pre><code># 118796 variants with DENTIST result 
# 120845 variants with susie_rss result 
# 118796 variants with both DENTIST and susie_rss result 
# 0 variants detected by both DENTIST and susie_rss (pvalue &lt; 5e-8).
# 
#            null      SuSiE only    DENTIST only SuSiE &amp; DENTIST 
#          118705              91               0               0</code></pre>
<pre class="r"><code>ggplot(df, aes(x = dentist.pval, y = susie.pval)) +
  geom_point(alpha=0.6) +
  xlim(0, 100) + ylim(0, 100) +
  labs(x = &quot;DENTIST -log10P&quot;, y = &quot;SuSiE RSS -log10P&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) +
  geom_abline(intercept = 0, slope = 1) + 
  geom_vline(xintercept = -log10(5e-8), col = &quot;red&quot;) +
  geom_hline(yintercept = -log10(5e-8), col = &quot;red&quot;) +
  theme_bw()</code></pre>
<pre><code># Warning: Removed 6 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-1.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df %&gt;% dplyr::mutate(susie.pval = ifelse(susie.pval &gt; 10, 10, susie.pval)), 
       aes(x = condmean, y = z, col = susie.pval)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;SuSiE -log10P \n(truncated at 10)&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-2">
Past versions of unnamed-chunk-8-2.png
</button>
</p>
<div id="fig-unnamed-chunk-8-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-2.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df,
       aes(x = condmean, y = z, col = dentist.pval)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;DENTIST -log10P&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-3">
Past versions of unnamed-chunk-8-3.png
</button>
</p>
<div id="fig-unnamed-chunk-8-3" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-3.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df, aes(x = condmean, y = z, col = type)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-4.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-4">
Past versions of unnamed-chunk-8-4.png
</button>
</p>
<div id="fig-unnamed-chunk-8-4" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-4.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df, aes(x = condmean, y = z, col = factor(allele_flipping))) +
  geom_point(alpha=0.6) +
  scale_colour_manual(values = c(&quot;0&quot; = &quot;grey&quot;, &quot;1&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value&quot;, y = &quot;Observed z scores from SuSiE&quot;, color = &quot;Possible allele flipping from SuSiE&quot;, 
       title = paste0(trait, &quot; chr&quot;, CHR)) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-8-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>Allele frequency</p>
<pre class="r"><code>p1 &lt;- ggplot(df[df$type == &quot;SuSiE only&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;green&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;SuSiE only&quot;) +
  theme_bw()

p2 &lt;- ggplot(df[df$type == &quot;null&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;grey&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;null&quot;) +
  theme_bw()

cowplot::plot_grid(p1, p2, ncol = 1)</code></pre>
<pre><code># `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-9-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-9-1.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Distribution of SuSiE only SNPs across loci</p>
<pre class="r"><code>locus_counts.df &lt;- data.frame(locus = unique(df$locus[df$type != &quot;null&quot;]), 
                              n_dentist_only = 0, 
                              n_susie_only = 0, 
                              n_dentist_susie = 0)

locus_counts &lt;- as.data.frame(table(df$locus[df$type == &quot;SuSiE only&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_susie_only &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

ggplot(locus_counts.df, aes(x = locus, y = n_susie_only)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;grey&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;SuSiE only&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-10-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-10-1.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code># locus_counts.df %&gt;% dplyr::arrange(desc(n_susie_only - n_dentist_only)) %&gt;% head()</code></pre>
</div>
<div id="example-locus" class="section level4">
<h4>Example locus</h4>
<pre class="r"><code>locus = &quot;950&quot;

region_df &lt;- regions_df[regions_df$locus == locus,]
print(region_df)
target_bp &lt;- round((region_df$start+region_df$stop)/2)
radius &lt;- max(abs(region_df$start - target_bp), abs(region_df$stop - target_bp))

cat(&quot;target_BP:&quot;, target_bp, &quot;radius:&quot;, radius, &quot;\n&quot;)

# ldref_res &lt;- load_UKBB_R_snp_info(region_df, &quot;/project2/mstephens/wcrouse/UKB_LDR_0.1&quot;, &quot;ukb_b38_0.1&quot;)
# length(ldref_res$R_snp_info$id)</code></pre>
<pre><code>#       chr    start     stop locus
# 950 chr22 24588236 26395662   950
# target_BP: 25491949 radius: 903713</code></pre>
<!-- load DENTIST results targeted at this locus -->
<!-- ```{r} -->
<!-- dentist.dir <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/", trait) -->
<!-- dentist.res.file <- file.path(dentist.dir, paste0("LDL-ukb-d-30780_irnt.ukb_chr", CHR, ".b38.target_25491949.radius_903713.niter10.DENTIST.full.txt")) -->
<!-- dentist.locus.df <- data.table::fread(dentist.res.file) -->
<!-- colnames(dentist.locus.df) <- c("rsID", "chisq", "LP", "ifDup") -->
<!-- dentist.locus.snps <- dentist.locus.df$rsID -->
<!-- dentist_detected_snps <- dentist.locus.df$rsID[which(dentist.locus.df$LP > -log10(5e-8))] -->
<!-- cat(sprintf("%d variants with DENTIST result in chr%s \n", length(dentist.locus.snps), CHR)) -->
<!-- cat(sprintf("%d detected variants with DENTIST pvalue < 5e-8.\n", length(dentist_detected_snps))) -->
<!-- ``` -->
<!-- Load SuSiE result for this locus -->
<!-- ```{r} -->
<!-- select_loci <- paste0("chr", CHR) -->
<!-- susie_rss_dir <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/", trait) -->
<!-- condz_dist_chr <- readRDS(file.path(susie_rss_dir, paste0(trait, ".condz.dist.", select_loci, "loci.rds"))) -->
<!-- condz_dist_locus.df <- condz_dist_chr[[locus]] -->
<!-- susierss.locus.snps <- condz_dist_locus.df$id -->
<!-- susierss_detected_snps <- condz_dist_locus.df$id[which(condz_dist_locus.df$p_diff < 5e-8)] -->
<!-- susierss_detected_flipped_snps <- condz_dist_locus.df$id[which(condz_dist_locus.df$logLR > 2 & abs(condz_dist_locus.df$z) > 2)] -->
<!-- cat(sprintf("%d variants with susie_rss result in chr%s \n", length(susierss.locus.snps), CHR)) -->
<!-- cat(sprintf("%d detected variants with susie_rss pvalue < 5e-8.\n", length(susierss_detected_snps))) -->
<!-- cat(sprintf("%d detected variants with susie_rss allele flipped.\n", length(susierss_detected_flipped_snps))) -->
<!-- ``` -->
<!-- Compare SuSiE RSS with DENTIST result in this locus -->
<!-- ```{r} -->
<!-- cat(sprintf("%d variants with DENTIST result \n", length(dentist.locus.snps))) -->
<!-- cat(sprintf("%d variants with susie_rss result \n", length(susierss.locus.snps))) -->
<!-- cat(sprintf("%d variants with both DENTIST and susie_rss result \n", length(intersect(dentist.locus.snps, susierss.locus.snps)))) -->
<!-- cat(sprintf("%d variants detected by both DENTIST and susie_rss (pvalue < 5e-8).\n", length(intersect(dentist_detected_snps, susierss_detected_snps)))) -->
<!-- common.snps <- intersect(dentist.locus.snps, susierss.locus.snps) -->
<!-- df_locus <- data.frame(rsID = common.snps, dentist.pval = NA, susie.pval = NA) -->
<!-- m.dentist <- match(common.snps, dentist.locus.df$rsID) -->
<!-- df_locus$dentist.pval <- dentist.locus.df$LP[m.dentist] -->
<!-- m.susie <- match(common.snps, condz_dist_locus.df$id) -->
<!-- df_locus$susie.pval <- -log10(condz_dist_locus.df$p_diff[m.susie]) -->
<!-- df_locus$z <- condz_dist_locus.df$z[m.susie] -->
<!-- df_locus$condmean <- condz_dist_locus.df$condmean[m.susie] -->
<!-- df_locus$logLR <- condz_dist_locus.df$logLR[m.susie] -->
<!-- df_locus$Freq_A1 <- dentist.chr.freq.df$Freq_A1[match(common.snps, dentist.chr.freq.df$RS_ID)] -->
<!-- df_locus$MAF <- pmin(df_locus$Freq_A1, 1-df_locus$Freq_A1) -->
<!-- m.sumstats <- match(common.snps, sumstats_hg38$snp) -->
<!-- df_locus$p <- sumstats_hg38$p[m.sumstats] -->
<!-- df_locus$locus <- sumstats_hg38$locus[m.sumstats] -->
<!-- df_locus$freq_A1_GWAS <- sumstats_hg38$freq[m.sumstats] -->
<!-- df_locus$MAF_GWAS <- pmin(df_locus$freq_A1_GWAS, 1-df_locus$freq_A1_GWAS) -->
<!-- df_locus <- df_locus %>% dplyr::mutate(allele_flipping = ifelse(logLR > 2 & abs(z) > 2, 1, 0)) -->
<!-- df_locus <- df_locus %>% dplyr::mutate(type = case_when(susie.pval > -log10(5e-8) & dentist.pval > -log10(5e-8) ~ "SuSiE & DENTIST",  -->
<!--                                       susie.pval > -log10(5e-8) ~ "SuSiE only", -->
<!--                                       dentist.pval > -log10(5e-8) ~ "DENTIST only", -->
<!--                                       .default = "null")) -->
<!-- df_locus$type <- factor(df_locus$type, levels = c("null",  "SuSiE only", "DENTIST only", "SuSiE & DENTIST")) -->
<!-- table(df_locus$type) -->
<!-- ``` -->
<pre class="r"><code>df_locus &lt;- df[df$locus == locus,]
# df_locus %&gt;% dplyr::arrange(desc(dentist.pval - susie.pval)) %&gt;% head()

# scatter plot comparing DENTIST vs. SuSiE pvalues
ggplot(df_locus, aes(x = dentist.pval, y = susie.pval)) +
  geom_point(alpha=0.6) +
  xlim(0, 50) + 
  # ylim(0, 50) +
  labs(x = &quot;DENTIST -log10P&quot;, y = &quot;SuSiE RSS -log10P&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) +
  geom_abline(intercept = 0, slope = 1) + 
  geom_vline(xintercept = -log10(5e-8), col = &quot;red&quot;) +
  geom_hline(yintercept = -log10(5e-8), col = &quot;red&quot;) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus %&gt;% dplyr::mutate(susie.pval = ifelse(susie.pval &gt; 10, 10, susie.pval)), 
       aes(x = condmean, y = z, col = susie.pval)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;SuSiE -log10P \n(truncated at 10)&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-12-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus,
       aes(x = condmean, y = z, col = dentist.pval)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;DENTIST -log10P&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-12-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus, aes(x = condmean, y = z, col = type)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-12-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggplot(df_locus, aes(x = condmean, y = z, col = factor(allele_flipping))) +
#   geom_point(alpha=0.6) +
#   geom_abline(intercept = 0, slope = 1) +
#   scale_colour_manual(values = c(&quot;0&quot; = &quot;grey&quot;, &quot;1&quot; = &quot;red&quot;)) +
#   labs(x = &quot;Expected value&quot;, y = &quot;Observed z scores from SuSiE&quot;,
#        color = &quot;Possible allele flipping from SuSiE&quot;,
#        title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) +
#   theme_bw()</code></pre>
</div>
</div>
<div id="genome-wide-results" class="section level3">
<h3>Genome-wide results</h3>
<p>DENTIST results genome-wide</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res &lt;- foreach(CHR=1:22) %do% {
  dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;LDL-ukb-d-30780_irnt.ukb_chr&quot;, CHR, &quot;.b38.DENTIST.full.txt&quot;))
  if(file.exists(dentist.res.file)){
    dentist.chr.res &lt;- data.table::fread(dentist.res.file)
    colnames(dentist.chr.res) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)
    cbind(chr = CHR, dentist.chr.res)
  }else{
    NULL
  }
}
dentist.res.df &lt;- do.call(rbind.data.frame, dentist.res)
data.table::fwrite(dentist.res.df, file.path(dentist.dir, paste0(&quot;LDL-ukb-d-30780_irnt.ukb_chrs.b38.DENTIST.full.txt.gz&quot;)), sep = &quot;\t&quot;, col.names = TRUE)</code></pre>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res.df &lt;- data.table::fread(file.path(dentist.dir, paste0(&quot;LDL-ukb-d-30780_irnt.ukb_chrs.b38.DENTIST.full.txt.gz&quot;)))
cat(sprintf(&quot;%d variants with DENTIST result in total \n&quot;, length(dentist.res.df$rsID)))
dentist_detected_snps &lt;- dentist.res.df$rsID[which(dentist.res.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d detected variants (%.3f%%) with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist_detected_snps), length(dentist_detected_snps)/length(dentist.res.df$rsID)*100))</code></pre>
<pre><code># 8687900 variants with DENTIST result in total 
# 292 detected variants (0.003%) with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>SuSiE RSS results genome-wide</p>
<pre class="r"><code>susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist.res &lt;- foreach(CHR=1:22) %do% {
  select_loci &lt;- paste0(&quot;chr&quot;, CHR)
  condz_dist.file &lt;- file.path(susie_rss_dir, paste0(trait, &quot;.condz.dist.&quot;, select_loci, &quot;loci.rds&quot;))
  if(file.exists(condz_dist.file)){
    condz_dist_chr &lt;- readRDS(condz_dist.file)
    condz_dist_chr.df &lt;- do.call(rbind.data.frame, condz_dist_chr)
    cbind(chr = CHR, condz_dist_chr.df)
  }else{
    NULL
  }
}
# summary(condz_dist.res)
condz_dist_all.df &lt;- do.call(rbind.data.frame, condz_dist.res)
saveRDS(condz_dist_all.df, file.path(susie_rss_dir, paste0(trait, &quot;.susie_rss.condz.dist.rds&quot;)))</code></pre>
<pre class="r"><code>susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist_all.df &lt;- readRDS(file.path(susie_rss_dir, paste0(trait, &quot;.susie_rss.condz.dist.rds&quot;)))
susierss_detected_snps &lt;- condz_dist_all.df$id[which(condz_dist_all.df$p_diff &lt; 5e-8)]
susierss_detected_flipped_snps &lt;- condz_dist_all.df$id[which(condz_dist_all.df$logLR &gt; 2 &amp; abs(condz_dist_all.df$z) &gt; 2)]
cat(sprintf(&quot;%d variants with susie_rss result in total \n&quot;, length(condz_dist_all.df$id)))
cat(sprintf(&quot;%d detected variants (%.3f%%) with susie_rss pvalue &lt; 5e-8.\n&quot;, length(susierss_detected_snps), length(susierss_detected_snps)/length(condz_dist_all.df$id)*100))
cat(sprintf(&quot;%d detected variants with susie_rss allele flipped.\n&quot;, length(susierss_detected_flipped_snps)))</code></pre>
<pre><code># 8841628 variants with susie_rss result in total 
# 4607 detected variants (0.052%) with susie_rss pvalue &lt; 5e-8.
# 8 detected variants with susie_rss allele flipped.</code></pre>
</div>
</div>
<div id="afib" class="section level2">
<h2>aFib</h2>
<pre class="r"><code>trait &lt;- &quot;aFib&quot;</code></pre>
<pre class="r"><code>sumstats &lt;- readRDS(&quot;/project2/xinhe/shared_data/multigroup_ctwas/gwas/gwas_processed/aFib-ebi-a-GCST006414.sumstats.RDS&quot;)
# sumstats$z &lt;- sumstats$beta/sumstats$se
# z_snp &lt;- sumstats[,c(&quot;ID&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;z&quot;, &quot;N&quot;)]
# colnames(z_snp) &lt;- c(&quot;id&quot;, &quot;A1&quot;, &quot;A2&quot;, &quot;z&quot;, &quot;ss&quot;)
# z_snp &lt;- z_snp[!(z_snp$id %in% z_snp$id[duplicated(z_snp$id)]),] #drop multiallelic variants (id not unique)

sumstats &lt;- makeGRangesFromDataFrame(sumstats, start.field = &quot;pos&quot;, end.field = &quot;pos&quot;, keep.extra.columns = T)
sumstats_hg38 &lt;- liftOver_hg19ToHg38(sumstats) %&gt;% as.data.frame() %&gt;%
    dplyr::rename(chr = seqnames, pos = start, snp = SNP) %&gt;%
    dplyr::select(chr, pos, snp, A1, A2, beta, se, p, freq)
sumstats_hg38 &lt;- mapgen::assign_snp_locus(sumstats_hg38, regions_df)</code></pre>
<div id="compare-dentist-and-susie-rss-pvalues-in-chr22-1" class="section level3">
<h3>Compare DENTIST and SuSiE RSS pvalues in chr22</h3>
<p>load DENTIST results</p>
<pre class="r"><code>CHR=22
dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.DENTIST.full.txt&quot;))
dentist.chr.df &lt;- data.table::fread(dentist.res.file)
colnames(dentist.chr.df) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)

dentist.chr.snps &lt;- dentist.chr.df$rsID
dentist_detected_snps &lt;- dentist.chr.df$rsID[which(dentist.chr.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d variants with DENTIST result in chr%s \n&quot;, length(dentist.chr.snps), CHR))
cat(sprintf(&quot;%d detected variants with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist_detected_snps)))

# dentist.chr.short.snps &lt;- read.table(file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.DENTIST.short.txt&quot;)))$V1
# length(dentist.chr.short.snps)
# setequal(dentist_detected_snps, dentist.chr.short.snps)</code></pre>
<pre><code># 109507 variants with DENTIST result in chr22 
# 2802 detected variants with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>load DENTIST results with one iteration</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.niter1.DENTIST.full.txt&quot;))
dentist.chr.df &lt;- data.table::fread(dentist.res.file)
colnames(dentist.chr.df) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)

dentist.chr.snps &lt;- dentist.chr.df$rsID
dentist_detected_snps &lt;- dentist.chr.df$rsID[which(dentist.chr.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d variants with DENTIST result in chr%s \n&quot;, length(dentist.chr.snps), CHR))
cat(sprintf(&quot;%d detected variants with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist_detected_snps)))

# dentist.chr.short.snps &lt;- read.table(file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.niter1.DENTIST.short.txt&quot;)))$V1
# length(dentist.chr.short.snps)
# setequal(dentist_detected_snps, dentist.chr.short.snps)</code></pre>
<pre><code># 109507 variants with DENTIST result in chr22 
# 2596 detected variants with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>Load Allele Frequency</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.chr.freq.df &lt;- data.table::fread(file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.frq&quot;)))</code></pre>
<p>load SuSiE RSS result</p>
<pre class="r"><code>select_loci &lt;- paste0(&quot;chr&quot;, CHR)
susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist_chr &lt;- readRDS(file.path(susie_rss_dir, paste0(trait, &quot;.condz.dist.&quot;, select_loci, &quot;loci.rds&quot;)))
condz_dist_chr.df &lt;- do.call(rbind.data.frame, condz_dist_chr)

susierss.chr.snps &lt;- condz_dist_chr.df$id
susierss_detected_snps &lt;- condz_dist_chr.df$id[which(condz_dist_chr.df$p_diff &lt; 5e-8)]
susierss_detected_flipped_snps &lt;- condz_dist_chr.df$id[which(condz_dist_chr.df$logLR &gt; 2 &amp; abs(condz_dist_chr.df$z) &gt; 2)]
cat(sprintf(&quot;%d variants with susie_rss result in chr%s \n&quot;, length(susierss.chr.snps), CHR))
cat(sprintf(&quot;%d detected variants with susie_rss pvalue &lt; 5e-8.\n&quot;, length(susierss_detected_snps)))
cat(sprintf(&quot;%d detected variants with susie_rss allele flipped.\n&quot;, length(susierss_detected_flipped_snps)))</code></pre>
<pre><code># 110716 variants with susie_rss result in chr22 
# 449 detected variants with susie_rss pvalue &lt; 5e-8.
# 12 detected variants with susie_rss allele flipped.</code></pre>
<div id="compare-susie-rss-with-dentist-result-one-iteration" class="section level4">
<h4>Compare SuSiE RSS with DENTIST result (one iteration)</h4>
<pre class="r"><code>cat(sprintf(&quot;%d variants with DENTIST result \n&quot;, length(dentist.chr.snps)))
cat(sprintf(&quot;%d variants with susie_rss result \n&quot;, length(susierss.chr.snps)))
cat(sprintf(&quot;%d variants with both DENTIST and susie_rss result \n&quot;, length(intersect(dentist.chr.snps, susierss.chr.snps))))
cat(sprintf(&quot;%d variants detected by both DENTIST and susie_rss (pvalue &lt; 5e-8).\n&quot;, length(intersect(dentist_detected_snps, susierss_detected_snps))))

common.snps &lt;- intersect(dentist.chr.snps, susierss.chr.snps)
df &lt;- data.frame(rsID = common.snps, dentist.pval = NA, susie.pval = NA)
m.dentist &lt;- match(common.snps, dentist.chr.df$rsID)
df$dentist.pval &lt;- dentist.chr.df$LP[m.dentist]
m.susie &lt;- match(common.snps, condz_dist_chr.df$id)
df$susie.pval &lt;- -log10(condz_dist_chr.df$p_diff[m.susie])
df$z &lt;- condz_dist_chr.df$z[m.susie]
df$condmean &lt;- condz_dist_chr.df$condmean[m.susie]
df$logLR &lt;- condz_dist_chr.df$logLR[m.susie]
df$Freq_A1 &lt;- dentist.chr.freq.df$Freq_A1[match(common.snps, dentist.chr.freq.df$RS_ID)]
df$MAF &lt;- pmin(df$Freq_A1, 1-df$Freq_A1)

m.sumstats &lt;- match(common.snps, sumstats_hg38$snp)
df$p &lt;- sumstats_hg38$p[m.sumstats]
df$locus &lt;- sumstats_hg38$locus[m.sumstats]
df$freq_A1_GWAS &lt;- sumstats_hg38$freq[m.sumstats]
df$MAF_GWAS &lt;- pmin(df$freq_A1_GWAS, 1-df$freq_A1_GWAS)

df &lt;- df %&gt;% dplyr::mutate(allele_flipping = ifelse(logLR &gt; 2 &amp; abs(z) &gt; 2, 1, 0))

df &lt;- df %&gt;% dplyr::mutate(type = case_when(susie.pval &gt; -log10(5e-8) &amp; dentist.pval &gt; -log10(5e-8) ~ &quot;SuSiE &amp; DENTIST&quot;, 
                                      susie.pval &gt; -log10(5e-8) ~ &quot;SuSiE only&quot;,
                                      dentist.pval &gt; -log10(5e-8) ~ &quot;DENTIST only&quot;,
                                      .default = &quot;null&quot;))

df$type &lt;- factor(df$type, levels = c(&quot;null&quot;,  &quot;SuSiE only&quot;, &quot;DENTIST only&quot;, &quot;SuSiE &amp; DENTIST&quot;))
table(df$type)</code></pre>
<pre><code># 109507 variants with DENTIST result 
# 110716 variants with susie_rss result 
# 109504 variants with both DENTIST and susie_rss result 
# 355 variants detected by both DENTIST and susie_rss (pvalue &lt; 5e-8).
# 
#            null      SuSiE only    DENTIST only SuSiE &amp; DENTIST 
#          106847              61            2241             355</code></pre>
<pre class="r"><code>ggplot(df, aes(x = dentist.pval, y = susie.pval)) +
  geom_point(alpha=0.6) +
  xlim(0, 100) + ylim(0, 100) +
  labs(x = &quot;DENTIST -log10P&quot;, y = &quot;SuSiE RSS -log10P&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) +
  geom_abline(intercept = 0, slope = 1) + 
  geom_vline(xintercept = -log10(5e-8), col = &quot;red&quot;) +
  geom_hline(yintercept = -log10(5e-8), col = &quot;red&quot;) +
  theme_bw()</code></pre>
<pre><code># Warning: Removed 25 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-25-1">
Past versions of unnamed-chunk-25-1.png
</button>
</p>
<div id="fig-unnamed-chunk-25-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-1.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df %&gt;% dplyr::mutate(susie.pval = ifelse(susie.pval &gt; 10, 10, susie.pval)), 
       aes(x = condmean, y = z, col = susie.pval)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;SuSiE -log10P \n(truncated at 10)&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df %&gt;% dplyr::mutate(dentist.pval = ifelse(dentist.pval &gt; 10, 10, dentist.pval)),
       aes(x = condmean, y = z, col = dentist.pval)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;DENTIST -log10P \n(truncated at 10)&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df, aes(x = condmean, y = z, col = type)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df, aes(x = condmean, y = z, col = factor(allele_flipping))) +
  geom_point(alpha=0.6) +
  scale_colour_manual(values = c(&quot;0&quot; = &quot;grey&quot;, &quot;1&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value&quot;, y = &quot;Observed z scores from SuSiE&quot;,
       color = &quot;Possible allele flipping from SuSiE&quot;, title = paste0(trait, &quot; chr&quot;, CHR)) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-25-5.png" width="672" style="display: block; margin: auto;" /></p>
<p>GWAS p-values</p>
<pre class="r"><code>ggplot(df, aes(x=type, y=-log10(p), fill=type)) +
  geom_boxplot(show.legend = FALSE) +
  scale_fill_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-26-1.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-1">
Past versions of unnamed-chunk-26-1.png
</button>
</p>
<div id="fig-unnamed-chunk-26-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-26-1.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>ggplot(df[df$type != &quot;null&quot;,], aes(x=type, y=-log10(p), fill=type)) +
  geom_boxplot(show.legend = FALSE) +
  scale_fill_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-26-2.png" width="480" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-26-2">
Past versions of unnamed-chunk-26-2.png
</button>
</p>
<div id="fig-unnamed-chunk-26-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a79e8529a5d0b146760295e320f15c1bd3640cb/docs/figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-26-2.png" target="_blank">0a79e85</a>
</td>
<td>
kevinlkx
</td>
<td>
2023-12-04
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Allele frequency</p>
<pre class="r"><code>ggplot(df, aes(x=type, y=MAF, fill=type)) +
  geom_boxplot(show.legend = FALSE) +
  scale_fill_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-27-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>p1 &lt;- ggplot(df[df$type == &quot;SuSiE only&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;green&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;SuSiE only&quot;) +
  theme_bw()

p2 &lt;- ggplot(df[df$type == &quot;DENTIST only&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;blue&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;DENTIST only&quot;) +
  theme_bw()

p3 &lt;- ggplot(df[df$type == &quot;SuSiE &amp; DENTIST&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;red&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;SuSiE &amp; DENTIST&quot;) +
  theme_bw()

p4 &lt;- ggplot(df[df$type == &quot;null&quot;,], aes(x = MAF)) +
  geom_histogram(show.legend = FALSE, fill = &quot;grey&quot;, color = &quot;black&quot;) +
  ggtitle(&quot;null&quot;) +
  theme_bw()

cowplot::plot_grid(p1, p2, p3, p4)</code></pre>
<pre><code># `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.
# `stat_bin()` using `bins = 30`. Pick better value with `binwidth`.</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-28-1.png" width="576" style="display: block; margin: auto;" /></p>
<p>Distribution of detected SNPs across loci</p>
<pre class="r"><code>locus_counts.df &lt;- data.frame(locus = unique(df$locus[df$type != &quot;null&quot;]), 
                              n_dentist_only = 0, 
                              n_susie_only = 0, 
                              n_dentist_susie = 0)

locus_counts &lt;- as.data.frame(table(df$locus[df$type == &quot;DENTIST only&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_dentist_only &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts &lt;- as.data.frame(table(df$locus[df$type == &quot;SuSiE only&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_susie_only &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts &lt;- as.data.frame(table(df$locus[df$type == &quot;SuSiE &amp; DENTIST&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_dentist_susie &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts.df[is.na(locus_counts.df)] &lt;- 0

p1 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_dentist_only)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;blue&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;DENTIST only&quot;) +
  theme_bw()

p2 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_susie_only)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;green&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;SuSiE only&quot;) +
  theme_bw()

p3 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_dentist_susie)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;red&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;SuSiE &amp; DENTIST&quot;) +
  theme_bw()

cowplot::plot_grid(p1, p2, p3, ncol = 1)</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-29-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>locus_counts.df %&gt;% dplyr::arrange(desc(n_dentist_only - n_susie_only)) %&gt;% head()</code></pre>
<pre><code>#   locus n_dentist_only n_susie_only n_dentist_susie
# 1   956            194            0              18
# 2   960            194            1              11
# 3   949            177            1               6
# 4   952            171            1              19
# 5   957            152            0              24
# 6   953            146            3              21</code></pre>
<p>SNPs with large difference between DENTIST and SuSiE p-values</p>
<pre class="r"><code>diff_df &lt;- df %&gt;% dplyr::filter(abs(dentist.pval - susie.pval) &gt; 10)
table(diff_df$type)

table(diff_df$locus[diff_df$type == &quot;DENTIST only&quot;])

locus_counts.df &lt;- data.frame(locus = unique(diff_df$locus[diff_df$type != &quot;null&quot;]), 
                              n_dentist_only = 0, 
                              n_susie_only = 0, 
                              n_dentist_susie = 0)

locus_counts &lt;- as.data.frame(table(diff_df$locus[diff_df$type == &quot;DENTIST only&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_dentist_only &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts &lt;- as.data.frame(table(diff_df$locus[diff_df$type == &quot;SuSiE only&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_susie_only &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts &lt;- as.data.frame(table(diff_df$locus[diff_df$type == &quot;SuSiE &amp; DENTIST&quot;]))
colnames(locus_counts) &lt;- c(&quot;locus&quot;, &quot;count&quot;)
locus_counts.df$n_dentist_susie &lt;- locus_counts$count[match(locus_counts.df$locus, locus_counts$locus)]

locus_counts.df[is.na(locus_counts.df)] &lt;- 0

p1 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_dentist_only)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;blue&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;DENTIST only&quot;) +
  theme_bw()

p2 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_susie_only)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;green&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;SuSiE only&quot;) +
  theme_bw()

p3 &lt;- ggplot(locus_counts.df, aes(x = locus, y = n_dentist_susie)) +
  geom_bar(stat = &quot;identity&quot;, fill = &quot;red&quot;) +
  scale_x_continuous(breaks=unique(locus_counts.df$locus), labels = unique(locus_counts.df$locus)) +
  labs(x = &quot;locus&quot;, y = &quot;number of SNPs&quot;) +
  ggtitle(&quot;SuSiE &amp; DENTIST&quot;) +
  theme_bw()

cowplot::plot_grid(p1, p2, p3, ncol = 1)</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-30-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># locus_counts.df %&gt;% dplyr::arrange(desc(n_dentist_only - n_susie_only)) %&gt;% head()</code></pre>
<pre><code># 
#            null      SuSiE only    DENTIST only SuSiE &amp; DENTIST 
#               0              12            1041             222 
# 
# 944 945 946 947 948 949 950 951 952 953 954 955 956 957 958 959 960 961 962 963 
#   4  13  58  28  47  91  26   6  73  75  76  45 144  29  31   6 111  17  24  41 
# 964 965 966 967 
#  21  11  24  40</code></pre>
</div>
<div id="example-locus-1" class="section level4">
<h4>Example locus</h4>
<pre class="r"><code>locus = &quot;956&quot;

region_df &lt;- regions_df[regions_df$locus == locus,]
print(region_df)
target_bp &lt;- round((region_df$start+region_df$stop)/2)
radius &lt;- max(abs(region_df$start - target_bp), abs(region_df$stop - target_bp))

cat(&quot;target_BP:&quot;, target_bp, &quot;radius:&quot;, radius, &quot;\n&quot;)

# ldref_res &lt;- load_UKBB_R_snp_info(region_df, &quot;/project2/mstephens/wcrouse/UKB_LDR_0.1&quot;, &quot;ukb_b38_0.1&quot;)
# length(ldref_res$R_snp_info$id)</code></pre>
<pre><code>#       chr    start     stop locus
# 956 chr22 34098925 35134992   956
# target_BP: 34616958 radius: 518034</code></pre>
<!-- load DENTIST results targeted at this locus -->
<!-- ```{r} -->
<!-- dentist.dir <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/", trait) -->
<!-- dentist.res.file <- file.path(dentist.dir, paste0("aFib-ebi-a-GCST006414.ukb_chr", CHR, ".b38.target_34616958.radius_518034.niter10.DENTIST.full.txt")) -->
<!-- dentist.locus.df <- data.table::fread(dentist.res.file) -->
<!-- colnames(dentist.locus.df) <- c("rsID", "chisq", "LP", "ifDup") -->
<!-- dentist.locus.snps <- dentist.locus.df$rsID -->
<!-- dentist_detected_snps <- dentist.locus.df$rsID[which(dentist.locus.df$LP > -log10(5e-8))] -->
<!-- cat(sprintf("%d variants with DENTIST result in chr%s \n", length(dentist.locus.snps), CHR)) -->
<!-- cat(sprintf("%d detected variants with DENTIST pvalue < 5e-8.\n", length(dentist_detected_snps))) -->
<!-- # dentist.locus.short.snps <- read.table(file.path(dentist.dir, paste0("aFib-ebi-a-GCST006414.ukb_chr", CHR, ".b38.target_34616958.radius_518034.niter10.DENTIST.short.txt")))$V1 -->
<!-- # length(dentist.locus.short.snps) -->
<!-- # setequal(dentist_detected_snps, dentist.locus.short.snps) -->
<!-- ``` -->
<!-- Load SuSiE result for this locus -->
<!-- ```{r} -->
<!-- select_loci <- paste0("chr", CHR) -->
<!-- susie_rss_dir <- paste0("/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/", trait) -->
<!-- condz_dist_chr <- readRDS(file.path(susie_rss_dir, paste0(trait, ".condz.dist.", select_loci, "loci.rds"))) -->
<!-- condz_dist_locus.df <- condz_dist_chr[[locus]] -->
<!-- susierss.locus.snps <- condz_dist_locus.df$id -->
<!-- susierss_detected_snps <- condz_dist_locus.df$id[which(condz_dist_locus.df$p_diff < 5e-8)] -->
<!-- susierss_detected_flipped_snps <- condz_dist_locus.df$id[which(condz_dist_locus.df$logLR > 2 & abs(condz_dist_locus.df$z) > 2)] -->
<!-- cat(sprintf("%d variants with susie_rss result in chr%s \n", length(susierss.locus.snps), CHR)) -->
<!-- cat(sprintf("%d detected variants with susie_rss pvalue < 5e-8.\n", length(susierss_detected_snps))) -->
<!-- cat(sprintf("%d detected variants with susie_rss allele flipped.\n", length(susierss_detected_flipped_snps))) -->
<!-- ``` -->
<!-- Compare SuSiE RSS with DENTIST result in this locus -->
<!-- ```{r} -->
<!-- cat(sprintf("%d variants with DENTIST result \n", length(dentist.locus.snps))) -->
<!-- cat(sprintf("%d variants with susie_rss result \n", length(susierss.locus.snps))) -->
<!-- cat(sprintf("%d variants with both DENTIST and susie_rss result \n", length(intersect(dentist.locus.snps, susierss.locus.snps)))) -->
<!-- cat(sprintf("%d variants detected by both DENTIST and susie_rss (pvalue < 5e-8).\n", length(intersect(dentist_detected_snps, susierss_detected_snps)))) -->
<!-- common.snps <- intersect(dentist.locus.snps, susierss.locus.snps) -->
<!-- df_locus <- data.frame(rsID = common.snps, dentist.pval = NA, susie.pval = NA) -->
<!-- m.dentist <- match(common.snps, dentist.locus.df$rsID) -->
<!-- df_locus$dentist.pval <- dentist.locus.df$LP[m.dentist] -->
<!-- m.susie <- match(common.snps, condz_dist_locus.df$id) -->
<!-- df_locus$susie.pval <- -log10(condz_dist_locus.df$p_diff[m.susie]) -->
<!-- df_locus$z <- condz_dist_locus.df$z[m.susie] -->
<!-- df_locus$condmean <- condz_dist_locus.df$condmean[m.susie] -->
<!-- df_locus$logLR <- condz_dist_locus.df$logLR[m.susie] -->
<!-- df_locus$Freq_A1 <- dentist.chr.freq.df$Freq_A1[match(common.snps, dentist.chr.freq.df$RS_ID)] -->
<!-- df_locus$MAF <- pmin(df_locus$Freq_A1, 1-df_locus$Freq_A1) -->
<!-- m.sumstats <- match(common.snps, sumstats_hg38$snp) -->
<!-- df_locus$p <- sumstats_hg38$p[m.sumstats] -->
<!-- df_locus$locus <- sumstats_hg38$locus[m.sumstats] -->
<!-- df_locus$freq_A1_GWAS <- sumstats_hg38$freq[m.sumstats] -->
<!-- df_locus$MAF_GWAS <- pmin(df_locus$freq_A1_GWAS, 1-df_locus$freq_A1_GWAS) -->
<!-- df_locus <- df_locus %>% dplyr::mutate(allele_flipping = ifelse(logLR > 2 & abs(z) > 2, 1, 0)) -->
<!-- df_locus <- df_locus %>% dplyr::mutate(type = case_when(susie.pval > -log10(5e-8) & dentist.pval > -log10(5e-8) ~ "SuSiE & DENTIST",  -->
<!--                                       susie.pval > -log10(5e-8) ~ "SuSiE only", -->
<!--                                       dentist.pval > -log10(5e-8) ~ "DENTIST only", -->
<!--                                       .default = "null")) -->
<!-- df_locus$type <- factor(df_locus$type, levels = c("null",  "SuSiE only", "DENTIST only", "SuSiE & DENTIST")) -->
<!-- table(df_locus$type) -->
<!-- ``` -->
<pre class="r"><code>df_locus &lt;- df[df$locus == locus,]
# df_locus %&gt;% dplyr::arrange(desc(dentist.pval - susie.pval)) %&gt;% head()

# scatter plot comparing DENTIST vs. SuSiE pvalues
ggplot(df_locus, aes(x = dentist.pval, y = susie.pval)) +
  geom_point(alpha=0.6) +
  xlim(0, 100) + ylim(0, 100) +
  labs(x = &quot;DENTIST -log10P&quot;, y = &quot;SuSiE RSS -log10P&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) +
  geom_abline(intercept = 0, slope = 1) + 
  geom_vline(xintercept = -log10(5e-8), col = &quot;red&quot;) +
  geom_hline(yintercept = -log10(5e-8), col = &quot;red&quot;) +
  theme_bw()</code></pre>
<pre><code># Warning: Removed 7 rows containing missing values (`geom_point()`).</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-32-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus, 
       aes(x = condmean, y = z, col = susie.pval)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;SuSiE -log10P&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-32-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus %&gt;% dplyr::mutate(dentist.pval = ifelse(dentist.pval &gt; 10, 10, dentist.pval)), 
       aes(x = condmean, y = z, col = dentist.pval)) +
  geom_point() +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_gradient(low = &quot;grey&quot;, high = &quot;blue&quot;, name = &quot;DENTIST -log10P \n(truncated at 10)&quot;) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-32-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ggplot(df_locus, aes(x = condmean, y = z, col = type)) +
  geom_point(alpha=0.6) +
  geom_abline(intercept = 0, slope = 1) + 
  scale_colour_manual(values = c(&quot;null&quot; = &quot;grey&quot;, &quot;SuSiE only&quot; = &quot;green&quot;, &quot;DENTIST only&quot; = &quot;blue&quot;,  &quot;SuSiE &amp; DENTIST&quot; = &quot;red&quot;)) +
  labs(x = &quot;Expected value from SuSiE&quot;, y = &quot;Observed z scores&quot;, title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) + 
  theme_bw()</code></pre>
<p><img src="figure/detect_LD_mismatch_gwas_UKBBref_DENTIST_susie_rss.Rmd/unnamed-chunk-32-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code># ggplot(df_locus, aes(x = condmean, y = z, col = factor(allele_flipping))) +
#   geom_point(alpha=0.6) +
#   geom_abline(intercept = 0, slope = 1) + 
#   scale_colour_manual(values = c(&quot;0&quot; = &quot;grey&quot;, &quot;1&quot; = &quot;red&quot;)) +
#   labs(x = &quot;Expected value&quot;, y = &quot;Observed z scores from SuSiE&quot;, 
#        color = &quot;Possible allele flipping from SuSiE&quot;, 
#        title = paste0(trait, &quot; chr&quot;, CHR, &quot; locus&quot;, locus)) +
#   theme_bw()</code></pre>
</div>
</div>
<div id="genome-wide-results-1" class="section level3">
<h3>Genome-wide results</h3>
<p>DENTIST results genome-wide</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res &lt;- foreach(CHR=1:22) %do% {
  dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.DENTIST.full.txt&quot;))
  if(file.exists(dentist.res.file)){
    dentist.chr.res &lt;- data.table::fread(dentist.res.file)
    colnames(dentist.chr.res) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)
    cbind(chr = CHR, dentist.chr.res)
  }else{
    NULL
  }
}
dentist.res.df &lt;- do.call(rbind.data.frame, dentist.res)
data.table::fwrite(dentist.res.df, file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chrs.b38.DENTIST.full.txt.gz&quot;)), sep = &quot;\t&quot;, col.names = TRUE)</code></pre>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res.df &lt;- data.table::fread(file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chrs.b38.DENTIST.full.txt.gz&quot;)))
cat(sprintf(&quot;%d variants with DENTIST result in total \n&quot;, length(dentist.res.df$rsID)))
dentist_detected_snps &lt;- dentist.res.df$rsID[which(dentist.res.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d detected variants (%.3f%%) with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist_detected_snps), length(dentist_detected_snps)/length(dentist.res.df$rsID)*100))</code></pre>
<pre><code># 8230059 variants with DENTIST result in total 
# 321808 detected variants (3.910%) with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>DENTIST 1 iteration results genome-wide</p>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.res &lt;- foreach(CHR=1:22) %do% {
  dentist.res.file &lt;- file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chr&quot;, CHR, &quot;.b38.niter1.DENTIST.full.txt&quot;))
  if(file.exists(dentist.res.file)){
    dentist.chr.res &lt;- data.table::fread(dentist.res.file)
    colnames(dentist.chr.res) &lt;- c(&quot;rsID&quot;, &quot;chisq&quot;, &quot;LP&quot;, &quot;ifDup&quot;)
    cbind(chr = CHR, dentist.chr.res)
  }else{
    NULL
  }
}
dentist.res.df &lt;- do.call(rbind.data.frame, dentist.res)
data.table::fwrite(dentist.res.df, file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chrs.b38.niter1.DENTIST.full.txt.gz&quot;)), sep = &quot;\t&quot;, col.names = TRUE)</code></pre>
<pre class="r"><code>dentist.dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/DENTIST/&quot;, trait)
dentist.1iter.res.df &lt;- data.table::fread(file.path(dentist.dir, paste0(&quot;aFib-ebi-a-GCST006414.ukb_chrs.b38.niter1.DENTIST.full.txt.gz&quot;)))
cat(sprintf(&quot;%d variants with DENTIST result in total \n&quot;, length(dentist.res.df$rsID)))
dentist.1iter_detected_snps &lt;- dentist.1iter.res.df$rsID[which(dentist.1iter.res.df$LP &gt; -log10(5e-8))]
cat(sprintf(&quot;%d detected variants (%.3f%%) with DENTIST pvalue &lt; 5e-8.\n&quot;, length(dentist.1iter_detected_snps), length(dentist.1iter_detected_snps)/length(dentist.1iter.res.df$rsID)*100))</code></pre>
<pre><code># 8230059 variants with DENTIST result in total 
# 307578 detected variants (3.737%) with DENTIST pvalue &lt; 5e-8.</code></pre>
<p>SuSiE RSS results genome-wide</p>
<pre class="r"><code>susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist.res &lt;- foreach(CHR=1:22) %do% {
  select_loci &lt;- paste0(&quot;chr&quot;, CHR)
  condz_dist.file &lt;- file.path(susie_rss_dir, paste0(trait, &quot;.condz.dist.&quot;, select_loci, &quot;loci.rds&quot;))
  if(file.exists(condz_dist.file)){
    condz_dist_chr &lt;- readRDS(condz_dist.file)
    condz_dist_chr.df &lt;- do.call(rbind.data.frame, condz_dist_chr)
    cbind(chr = CHR, condz_dist_chr.df)
  }else{
    NULL
  }
}
# summary(condz_dist.res)
condz_dist_all.df &lt;- do.call(rbind.data.frame, condz_dist.res)
saveRDS(condz_dist_all.df, file.path(susie_rss_dir, paste0(trait, &quot;.susie_rss.condz.dist.rds&quot;)))</code></pre>
<pre class="r"><code>susie_rss_dir &lt;- paste0(&quot;/project2/xinhe/shared_data/multigroup_ctwas/ld_mismatch_susie_rss/&quot;, trait)
condz_dist_all.df &lt;- readRDS(file.path(susie_rss_dir, paste0(trait, &quot;.susie_rss.condz.dist.rds&quot;)))
susierss_detected_snps &lt;- condz_dist_all.df$id[which(condz_dist_all.df$p_diff &lt; 5e-8)]
susierss_detected_flipped_snps &lt;- condz_dist_all.df$id[which(condz_dist_all.df$logLR &gt; 2 &amp; abs(condz_dist_all.df$z) &gt; 2)]
cat(sprintf(&quot;%d variants with susie_rss result in total \n&quot;, length(condz_dist_all.df$id)))
cat(sprintf(&quot;%d detected variants (%.3f%%) with susie_rss pvalue &lt; 5e-8.\n&quot;, length(susierss_detected_snps), length(susierss_detected_snps)/length(condz_dist_all.df$id)*100))
cat(sprintf(&quot;%d detected variants with susie_rss allele flipped.\n&quot;, length(susierss_detected_flipped_snps)))</code></pre>
<pre><code># 8258166 variants with susie_rss result in total 
# 38330 detected variants (0.464%) with susie_rss pvalue &lt; 5e-8.
# 439 detected variants with susie_rss allele flipped.</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code># R version 4.2.0 (2022-04-22)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: CentOS Linux 7 (Core)
# 
# Matrix products: default
# BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
# [1] stats4    stats     graphics  grDevices utils     datasets  methods  
# [8] base     
# 
# other attached packages:
#  [1] rtracklayer_1.58.0   GenomicRanges_1.48.0 GenomeInfoDb_1.34.9 
#  [4] IRanges_2.32.0       S4Vectors_0.36.1     BiocGenerics_0.44.0 
#  [7] forcats_1.0.0        stringr_1.5.0        dplyr_1.1.0         
# [10] purrr_1.0.1          readr_2.1.4          tidyr_1.3.0         
# [13] tibble_3.1.8         ggplot2_3.4.1        tidyverse_1.3.2     
# [16] data.table_1.14.6    foreach_1.5.2        susieR_0.12.35      
# [19] ctwas_0.1.35         workflowr_1.7.0     
# 
# loaded via a namespace (and not attached):
#   [1] googledrive_2.0.0           colorspace_2.1-0           
#   [3] rjson_0.2.21                ellipsis_0.3.2             
#   [5] rprojroot_2.0.3             XVector_0.38.0             
#   [7] fs_1.6.1                    rstudioapi_0.14            
#   [9] farver_2.1.1                fansi_1.0.4                
#  [11] lubridate_1.9.2             xml2_1.3.3                 
#  [13] R.methodsS3_1.8.2           codetools_0.2-18           
#  [15] logging_0.10-108            cachem_1.0.6               
#  [17] knitr_1.42                  jsonlite_1.8.4             
#  [19] Rsamtools_2.12.0            broom_1.0.3                
#  [21] dbplyr_2.3.0                R.oo_1.25.0                
#  [23] compiler_4.2.0              httr_1.4.4                 
#  [25] backports_1.4.1             assertthat_0.2.1           
#  [27] Matrix_1.5-3                fastmap_1.1.0              
#  [29] gargle_1.3.0                cli_3.6.0                  
#  [31] later_1.3.0                 htmltools_0.5.4            
#  [33] tools_4.2.0                 gtable_0.3.1               
#  [35] glue_1.6.2                  GenomeInfoDbData_1.2.9     
#  [37] Rcpp_1.0.10                 Biobase_2.58.0             
#  [39] cellranger_1.1.0            jquerylib_0.1.4            
#  [41] vctrs_0.5.2                 Biostrings_2.66.0          
#  [43] iterators_1.0.14            xfun_0.37                  
#  [45] plyranges_1.18.0            ps_1.7.2                   
#  [47] rvest_1.0.3                 timechange_0.2.0           
#  [49] lifecycle_1.0.3             irlba_2.3.5                
#  [51] mapgen_0.5.7.9000           restfulr_0.0.15            
#  [53] XML_3.99-0.13               googlesheets4_1.0.1        
#  [55] getPass_0.2-2               zlibbioc_1.44.0            
#  [57] scales_1.2.1                MatrixGenerics_1.10.0      
#  [59] hms_1.1.2                   promises_1.2.0.1           
#  [61] parallel_4.2.0              SummarizedExperiment_1.28.0
#  [63] yaml_2.3.7                  sass_0.4.5                 
#  [65] reshape_0.8.9               stringi_1.7.12             
#  [67] highr_0.10                  BiocIO_1.8.0               
#  [69] BiocParallel_1.32.5         rlang_1.0.6                
#  [71] pkgconfig_2.0.3             matrixStats_0.63.0         
#  [73] bitops_1.0-7                evaluate_0.20              
#  [75] lattice_0.20-45             labeling_0.4.2             
#  [77] GenomicAlignments_1.34.0    cowplot_1.1.1              
#  [79] processx_3.8.0              tidyselect_1.2.0           
#  [81] plyr_1.8.7                  magrittr_2.0.3             
#  [83] R6_2.5.1                    generics_0.1.3             
#  [85] DelayedArray_0.24.0         DBI_1.1.3                  
#  [87] pgenlibr_0.3.3              pillar_1.8.1               
#  [89] haven_2.5.1                 whisker_0.4                
#  [91] withr_2.5.0                 RCurl_1.98-1.10            
#  [93] mixsqp_0.3-43               modelr_0.1.10              
#  [95] crayon_1.5.2                utf8_1.2.3                 
#  [97] tzdb_0.3.0                  rmarkdown_2.20             
#  [99] grid_4.2.0                  readxl_1.4.2               
# [101] callr_3.7.3                 git2r_0.30.1               
# [103] reprex_2.0.2                digest_0.6.31              
# [105] httpuv_1.6.5                R.utils_2.12.2             
# [107] munsell_0.5.0               bslib_0.4.2</code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
