<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="shengqian" />

<meta name="date" content="2023-12-30" />

<title>Seven Correlated Tissues Simulation</title>

<script src="site_libs/header-attrs-2.19/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multigroup_ctwas_analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/sq-96/multigroup_ctwas_analysis">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Seven Correlated Tissues Simulation</h1>
<h4 class="author">shengqian</h4>
<h4 class="date">2023-12-30</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-01-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>multigroup_ctwas_analysis/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20231112code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20231112)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20231112code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20231112)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree2f6c0ddf06160fac3acc19edf5993a2cf12b93fatargetblank2f6c0dda"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa" target="_blank">2f6c0dd</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomsq96multigroupctwasanalysistree2f6c0ddf06160fac3acc19edf5993a2cf12b93fatargetblank2f6c0dda" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/sq-96/multigroup_ctwas_analysis/tree/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa" target="_blank">2f6c0dd</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Untracked files:
    Untracked:  analysis/LDL_Liver_Adipose_E_S.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Modified:   analysis/simulation_seven_tissues_correlated.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/simulation_seven_tissues_correlated.Rmd</code>) and HTML (<code>docs/simulation_seven_tissues_correlated.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa/analysis/simulation_seven_tissues_correlated.Rmd" target="_blank">2f6c0dd</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa/docs/simulation_seven_tissues_correlated.html" target="_blank">2f6c0dd</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
<td>
update
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/analysis/simulation_seven_tissues_correlated.Rmd" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
<td>
update
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/sq-96/multigroup_ctwas_analysis/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/simulation_seven_tissues_correlated.html" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
<td>
update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>library(ctwas)
library(data.table)
source(&quot;/project/xinhe/shengqian/cTWAS_simulation/summarize_basic_plots.R&quot;)</code></pre>
<pre><code>
Attaching package: &#39;ggpubr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:cowplot&#39;:

    get_legend</code></pre>
<pre class="r"><code>source(&quot;/project/xinhe/shengqian/cTWAS_simulation/summarize_ctwas_plots.R&quot;)</code></pre>
<pre><code>
Attaching package: &#39;plyr&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:ggpubr&#39;:

    mutate</code></pre>
<pre class="r"><code>source(&quot;/project2/xinhe/shengqian/cTWAS/cTWAS_analysis/analysis/simulation_help_functions.R&quot;)
plot_PIP &lt;- function(configtag, runtag,  simutags, ...){
   phenofs &lt;- paste0(results_dir, runtag, &quot;_simu&quot;, simutags, &quot;-pheno.Rd&quot;)
   susieIfs &lt;- paste0(results_dir, runtag, &quot;_simu&quot;,simutags, &quot;_config&quot;, configtag,&quot;_LDR&quot;,&quot;.susieIrss.txt&quot;)
   f1 &lt;- caliPIP_plot(phenofs, susieIfs, ...) 
   return(f1)
}</code></pre>
<div id="simulation-1-three-causal-tissues-and-four-non-causal-tissues" class="section level2">
<h2>Simulation 1: Three causal tissues and four non-causal tissues</h2>
<p>30% PVE and 2.5e-4 prior inclusion for SNPs, 3% PVE and 0.009 prior inclusion for Brain Cerebellum, Brain Hippocampus and Brain Caudate basal ganglia. 0% PVE for other tissues.</p>
<div id="shared-effect-size-parameters" class="section level3">
<h3>Shared effect size parameters</h3>
<p>For the cTWAS analysis, tissues shared an effect size parameter.</p>
<pre class="r"><code>results_dir &lt;- &quot;/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/&quot;
runtag = &quot;ukb-s80.45-3_7corr&quot;
configtag &lt;- 1

simutags &lt;- paste(1, 1:5, sep = &quot;-&quot;)
thin &lt;- 0.1

sample_size &lt;- 45000
PIP_threshold &lt;- 0.8</code></pre>
<pre class="r"><code>results_df &lt;- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)</code></pre>
<pre class="r"><code>#results using PIP threshold (gene+tissue)
results_df[,c(&quot;simutag&quot;, &quot;n_causal&quot;, &quot;n_detected_pip&quot;, &quot;n_detected_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal n_detected_pip n_detected_pip_in_causal
1     1-1      211             10                        8
2     1-2      233             32                       28
3     1-3      215             21                       18
4     1-4      212             30                       29
5     1-5      254             17                       16</code></pre>
<pre class="r"><code>#mean percent causal using PIP &gt; 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)</code></pre>
<pre><code>[1] 0.9</code></pre>
<pre class="r"><code>#results using combined PIP threshold
results_df[,c(&quot;simutag&quot;, &quot;n_causal_combined&quot;, &quot;n_detected_comb_pip&quot;, &quot;n_detected_comb_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal_combined n_detected_comb_pip n_detected_comb_pip_in_causal
1     1-1               209                  40                            36
2     1-2               232                  69                            55
3     1-3               212                  40                            37
4     1-4               212                  56                            52
5     1-5               253                  49                            43</code></pre>
<pre class="r"><code>#mean percent causal using combined PIP &gt; 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)</code></pre>
<pre><code>[1] 0.8779528</code></pre>
<pre class="r"><code>#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df))))]</code></pre>
<pre><code>  simutag    prior_SNP prior_Brain_Cerebellum prior_Brain_Hippocampus
1     1-1 0.0002395754            0.004483670            0.0068481880
2     1-2 0.0002427045            0.007869741            0.0132591028
3     1-3 0.0002355897            0.006018338            0.0003961507
4     1-4 0.0002386003            0.004966132            0.0113152506
5     1-5 0.0002455809            0.005096201            0.0106738503
  prior_Brain_Caudate_basal_ganglia prior_Brain_Cerebellar_Hemisphere
1                       0.000748970                      4.543431e-03
2                       0.012492771                      1.787225e-06
3                       0.012111091                      4.935098e-04
4                       0.007530475                      3.052034e-04
5                       0.009940482                      1.602602e-03
  prior_Brain_Cortex prior_Brain_Hypothalamus prior_Brain_Putamen_basal_ganglia
1       2.754072e-06             0.0034925487                      0.0026968720
2       1.488891e-03             0.0025931239                      0.0003692669
3       2.332537e-03             0.0009549596                      0.0020164558
4       1.487342e-05             0.0002722532                      0.0051363722
5       4.787281e-03             0.0031010995                      0.0003623928</code></pre>
<pre class="r"><code>colMeans(results_df[,setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df)))])</code></pre>
<pre><code>                        prior_SNP            prior_Brain_Cerebellum 
                     0.0002404102                      0.0056868163 
          prior_Brain_Hippocampus prior_Brain_Caudate_basal_ganglia 
                     0.0084985085                      0.0085647577 
prior_Brain_Cerebellar_Hemisphere                prior_Brain_Cortex 
                     0.0013893067                      0.0017252673 
         prior_Brain_Hypothalamus prior_Brain_Putamen_basal_ganglia 
                     0.0020827970                      0.0021162719 </code></pre>
<pre class="r"><code>#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;prior_var&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag prior_var_SNP prior_var_Brain_Cerebellum prior_var_Brain_Hippocampus
1     1-1      8.664607                   22.16341                    22.16341
2     1-2      8.468107                   17.40351                    17.40351
3     1-3      8.053270                   15.65865                    15.65865
4     1-4      9.213080                   16.77674                    16.77674
5     1-5      8.160598                   16.02897                    16.02897
  prior_var_Brain_Caudate_basal_ganglia prior_var_Brain_Cerebellar_Hemisphere
1                              22.16341                              22.16341
2                              17.40351                              17.40351
3                              15.65865                              15.65865
4                              16.77674                              16.77674
5                              16.02897                              16.02897
  prior_var_Brain_Cortex prior_var_Brain_Hypothalamus
1               22.16341                     22.16341
2               17.40351                     17.40351
3               15.65865                     15.65865
4               16.77674                     16.77674
5               16.02897                     16.02897
  prior_var_Brain_Putamen_basal_ganglia
1                              22.16341
2                              17.40351
3                              15.65865
4                              16.77674
5                              16.02897</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;prior_var&quot;, names(results_df))])</code></pre>
<pre><code>                        prior_var_SNP            prior_var_Brain_Cerebellum 
                             8.511932                             17.606253 
          prior_var_Brain_Hippocampus prior_var_Brain_Caudate_basal_ganglia 
                            17.606253                             17.606253 
prior_var_Brain_Cerebellar_Hemisphere                prior_var_Brain_Cortex 
                            17.606253                             17.606253 
         prior_var_Brain_Hypothalamus prior_var_Brain_Putamen_basal_ganglia 
                            17.606253                             17.606253 </code></pre>
<pre class="r"><code>#PVE and mean PVE
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;pve&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag   pve_SNP pve_Brain_Cerebellum pve_Brain_Hippocampus
1     1-1 0.2443557           0.01942860           0.026851422
2     1-2 0.2419332           0.02677741           0.040823104
3     1-3 0.2233366           0.01842478           0.001097412
4     1-4 0.2587660           0.01628910           0.033583565
5     1-5 0.2359109           0.01597068           0.030267860
  pve_Brain_Caudate_basal_ganglia pve_Brain_Cerebellar_Hemisphere
1                     0.003170917                    1.911470e-02
2                     0.041531674                    5.904227e-06
3                     0.036226089                    1.466889e-03
4                     0.024133137                    9.719500e-04
5                     0.030436649                    4.876164e-03
  pve_Brain_Cortex pve_Brain_Hypothalamus pve_Brain_Putamen_basal_ganglia
1     1.189866e-05           0.0137061600                     0.011161401
2     5.051098e-03           0.0079909213                     0.001200049
3     7.119819e-03           0.0026477446                     0.005896100
4     4.864122e-05           0.0008087556                     0.016091104
5     1.495824e-02           0.0088015269                     0.001084693</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;pve&quot;, names(results_df))])</code></pre>
<pre><code>                        pve_SNP            pve_Brain_Cerebellum 
                    0.240860492                     0.019378117 
          pve_Brain_Hippocampus pve_Brain_Caudate_basal_ganglia 
                    0.026524673                     0.027099693 
pve_Brain_Cerebellar_Hemisphere                pve_Brain_Cortex 
                    0.005287121                     0.005437939 
         pve_Brain_Hypothalamus pve_Brain_Putamen_basal_ganglia 
                    0.006791022                     0.007086669 </code></pre>
<pre class="r"><code>#TWAS results
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;twas&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag n_detected_twas n_detected_twas_in_causal n_detected_comb_twas
1     1-1             545                        58                  190
2     1-2             830                        72                  275
3     1-3             494                        61                  168
4     1-4             701                        70                  246
5     1-5             535                        56                  184
  n_detected_comb_twas_in_causal
1                             62
2                             75
3                             60
4                             72
5                             58</code></pre>
<pre class="r"><code>sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)</code></pre>
<pre><code>[1] 0.3076199</code></pre>
<pre class="r"><code>y1 &lt;- results_df$prior_Brain_Cerebellum
y2 &lt;- results_df$prior_Brain_Hippocampus
y3 &lt;- results_df$prior_Brain_Caudate_basal_ganglia
y4 &lt;- results_df$prior_Brain_Cerebellar_Hemisphere
y5 &lt;- results_df$prior_Brain_Cortex
y6 &lt;- results_df$prior_Brain_Hypothalamus
y7 &lt;- results_df$prior_Brain_Putamen_basal_ganglia


truth &lt;- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est &lt;- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c(&quot;Cerebellum&quot;,&quot;Hippocampus&quot;,&quot;Caudate_basal_ganglia&quot;,&quot;Cerebellar_Hemisphere&quot;,&quot;Cortex&quot;,&quot;Hypothalamus&quot;,&quot;Putamen_basal_ganglia&quot;),ylim=c(0,0.025),ylab=&quot;Prior inclusion&quot;)</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-5-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>y1 &lt;- results_df$pve_Brain_Cerebellum
y2 &lt;- results_df$pve_Brain_Hippocampus
y3 &lt;- results_df$pve_Brain_Caudate_basal_ganglia
y4 &lt;- results_df$pve_Brain_Cerebellar_Hemisphere
y5 &lt;- results_df$pve_Brain_Cortex
y6 &lt;- results_df$pve_Brain_Hypothalamus
y7 &lt;- results_df$pve_Brain_Putamen_basal_ganglia


truth &lt;- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est &lt;- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c(&quot;Cerebellum&quot;,&quot;Hippocampus&quot;,&quot;Caudate_basal_ganglia&quot;,&quot;Cerebellar_Hemisphere&quot;,&quot;Cortex&quot;,&quot;Hypothalamus&quot;,&quot;Putamen_basal_ganglia&quot;),ylim=c(0,0.06),ylab=&quot;PVE&quot;)</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-6-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>f1 &lt;- plot_PIP(configtag, runtag, paste(1, 1:5, sep = &quot;-&quot;), main = &quot;&quot;)</code></pre>
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre><code>Warning: The `size` argument of `element_line()` is deprecated as of ggplot2 3.4.0.
ℹ Please use the `linewidth` argument instead.
This warning is displayed once every 8 hours.
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>f1</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-7-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-7-1.png" target="_blank">2f6c0dd</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-7-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="separate-effect-size-parameters" class="section level3">
<h3>Separate effect size parameters</h3>
<p>For the cTWAS analysis, each tissue had its own prior inclusion parameter and effect size parameter.</p>
<pre class="r"><code>results_dir &lt;- &quot;/project/xinhe/shengqian/cTWAS_simulation/simulation_correlated_seven_tissues/&quot;
runtag = &quot;ukb-s80.45-3_7corr&quot;
configtag &lt;- 2

simutags &lt;- paste(1, 1:5, sep = &quot;-&quot;)
thin &lt;- 0.1

sample_size &lt;- 45000
PIP_threshold &lt;- 0.8</code></pre>
<pre class="r"><code>results_df &lt;- get_sim_joint_res(results_dir,runtag,configtag,simutags,thin,sample_size,PIP_threshold)</code></pre>
<pre class="r"><code>#results using PIP threshold (gene+tissue)
results_df[,c(&quot;simutag&quot;, &quot;n_causal&quot;, &quot;n_detected_pip&quot;, &quot;n_detected_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal n_detected_pip n_detected_pip_in_causal
1     1-1      211             11                        8
2     1-2      233             33                       29
3     1-3      215             24                       21
4     1-4      212             30                       29
5     1-5      254             16                       15</code></pre>
<pre class="r"><code>#mean percent causal using PIP &gt; 0.8
sum(results_df$n_detected_pip_in_causal)/sum(results_df$n_detected_pip)</code></pre>
<pre><code>[1] 0.8947368</code></pre>
<pre class="r"><code>#results using combined PIP threshold
results_df[,c(&quot;simutag&quot;, &quot;n_causal_combined&quot;, &quot;n_detected_comb_pip&quot;, &quot;n_detected_comb_pip_in_causal&quot;)]</code></pre>
<pre><code>  simutag n_causal_combined n_detected_comb_pip n_detected_comb_pip_in_causal
1     1-1               209                  41                            36
2     1-2               232                  72                            57
3     1-3               212                  41                            38
4     1-4               212                  54                            50
5     1-5               253                  49                            41</code></pre>
<pre class="r"><code>#mean percent causal using combined PIP &gt; 0.8
sum(results_df$n_detected_comb_pip_in_causal)/sum(results_df$n_detected_comb_pip)</code></pre>
<pre><code>[1] 0.8638132</code></pre>
<pre class="r"><code>#prior inclusion and mean prior inclusion
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df))))]</code></pre>
<pre><code>  simutag    prior_SNP prior_Brain_Cerebellum prior_Brain_Hippocampus
1     1-1 0.0002388188            0.004553117            0.0067204279
2     1-2 0.0002383224            0.008350487            0.0121854702
3     1-3 0.0002294211            0.006668920            0.0003872109
4     1-4 0.0002357946            0.003410703            0.0112904059
5     1-5 0.0002320507            0.003101128            0.0205486027
  prior_Brain_Caudate_basal_ganglia prior_Brain_Cerebellar_Hemisphere
1                       0.001041899                      0.0043737070
2                       0.012236224                      0.0005124543
3                       0.010046520                      0.0016055117
4                       0.008177793                      0.0009211086
5                       0.009264066                      0.0009420104
  prior_Brain_Cortex prior_Brain_Hypothalamus prior_Brain_Putamen_basal_ganglia
1        0.000484178             0.0037810618                       0.002623404
2        0.002434187             0.0019239138                       0.003480012
3        0.005639370             0.0015970038                       0.002961637
4        0.000370470             0.0003570503                       0.006688261
5        0.004062700             0.0049716004                       0.002879252</code></pre>
<pre class="r"><code>colMeans(results_df[,setdiff(grep(&quot;prior&quot;, names(results_df)), grep(&quot;prior_var&quot;, names(results_df)))])</code></pre>
<pre><code>                        prior_SNP            prior_Brain_Cerebellum 
                     0.0002348815                      0.0052168710 
          prior_Brain_Hippocampus prior_Brain_Caudate_basal_ganglia 
                     0.0102264235                      0.0081533005 
prior_Brain_Cerebellar_Hemisphere                prior_Brain_Cortex 
                     0.0016709584                      0.0025981809 
         prior_Brain_Hypothalamus prior_Brain_Putamen_basal_ganglia 
                     0.0025261260                      0.0037265131 </code></pre>
<pre class="r"><code>#prior variance and mean prior variance
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;prior_var&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag prior_var_SNP prior_var_Brain_Cerebellum prior_var_Brain_Hippocampus
1     1-1      8.683177                   21.22213                   23.507837
2     1-2      8.594290                   14.60871                   19.860611
3     1-3      8.182719                   12.49572                   43.785247
4     1-4      9.302851                   30.96215                   16.422893
5     1-5      8.539146                   31.75806                    5.129005
  prior_var_Brain_Caudate_basal_ganglia prior_var_Brain_Cerebellar_Hemisphere
1                              14.85855                             25.591868
2                              17.92465                              0.862841
3                              21.65575                              4.287568
4                              14.25179                              4.583603
5                              17.58322                             43.927529
  prior_var_Brain_Cortex prior_var_Brain_Hypothalamus
1              0.8232463                    15.015413
2              4.6919025                    29.792984
3              6.3542669                     5.761578
4              2.6826180                    14.362451
5             29.4854636                     5.940242
  prior_var_Brain_Putamen_basal_ganglia
1                             25.579367
2                              3.511273
3                              9.979775
4                             11.867409
5                              3.332750</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;prior_var&quot;, names(results_df))])</code></pre>
<pre><code>                        prior_var_SNP            prior_var_Brain_Cerebellum 
                             8.660437                             22.209353 
          prior_var_Brain_Hippocampus prior_var_Brain_Caudate_basal_ganglia 
                            21.741118                             17.254791 
prior_var_Brain_Cerebellar_Hemisphere                prior_var_Brain_Cortex 
                            15.850682                              8.807499 
         prior_var_Brain_Hypothalamus prior_var_Brain_Putamen_basal_ganglia 
                            14.174534                             10.854115 </code></pre>
<pre class="r"><code>#PVE and mean PVE
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;pve&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag   pve_SNP pve_Brain_Cerebellum pve_Brain_Hippocampus
1     1-1 0.2441061           0.01889163           0.027948899
2     1-2 0.2411050           0.02385036           0.042814415
3     1-3 0.2209848           0.01629251           0.002999373
4     1-4 0.2582150           0.02064650           0.032803057
5     1-5 0.2332538           0.01925506           0.018645348
  pve_Brain_Caudate_basal_ganglia pve_Brain_Cerebellar_Hemisphere
1                     0.002957235                    2.124705e-02
2                     0.041896900                    8.393305e-05
3                     0.041559736                    1.306687e-03
4                     0.022263289                    8.014286e-04
5                     0.031116012                    7.854879e-03
  pve_Brain_Cortex pve_Brain_Hypothalamus pve_Brain_Putamen_basal_ganglia
1     7.769999e-05           0.0100528192                     0.012530743
2     2.226328e-03           0.0101493080                     0.002281746
3     6.985253e-03           0.0016292369                     0.005519178
4     1.937305e-04           0.0009080186                     0.014821460
5     2.335118e-02           0.0052292231                     0.001791861</code></pre>
<pre class="r"><code>colMeans(results_df[,grep(&quot;pve&quot;, names(results_df))])</code></pre>
<pre><code>                        pve_SNP            pve_Brain_Cerebellum 
                    0.239532928                     0.019787215 
          pve_Brain_Hippocampus pve_Brain_Caudate_basal_ganglia 
                    0.025042218                     0.027958635 
pve_Brain_Cerebellar_Hemisphere                pve_Brain_Cortex 
                    0.006258796                     0.006566838 
         pve_Brain_Hypothalamus pve_Brain_Putamen_basal_ganglia 
                    0.005593721                     0.007388997 </code></pre>
<pre class="r"><code>#TWAS results
results_df[,c(which(colnames(results_df)==&quot;simutag&quot;), grep(&quot;twas&quot;, names(results_df)))]</code></pre>
<pre><code>  simutag n_detected_twas n_detected_twas_in_causal n_detected_comb_twas
1     1-1             545                        58                  190
2     1-2             830                        72                  275
3     1-3             494                        61                  168
4     1-4             701                        70                  246
5     1-5             535                        56                  184
  n_detected_comb_twas_in_causal
1                             62
2                             75
3                             60
4                             72
5                             58</code></pre>
<pre class="r"><code>sum(results_df$n_detected_comb_twas_in_causal)/sum(results_df$n_detected_comb_twas)</code></pre>
<pre><code>[1] 0.3076199</code></pre>
<pre class="r"><code>y1 &lt;- results_df$prior_Brain_Cerebellum
y2 &lt;- results_df$prior_Brain_Hippocampus
y3 &lt;- results_df$prior_Brain_Caudate_basal_ganglia
y4 &lt;- results_df$prior_Brain_Cerebellar_Hemisphere
y5 &lt;- results_df$prior_Brain_Cortex
y6 &lt;- results_df$prior_Brain_Hypothalamus
y7 &lt;- results_df$prior_Brain_Putamen_basal_ganglia


truth &lt;- rbind(c(1,0.009),c(2,0.009),c(3,0.009),c(4,0),c(5,0),c(6,0),c(7,0))
est &lt;- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c(&quot;Cerebellum&quot;,&quot;Hippocampus&quot;,&quot;Caudate_basal_ganglia&quot;,&quot;Cerebellar_Hemisphere&quot;,&quot;Cortex&quot;,&quot;Hypothalamus&quot;,&quot;Putamen_basal_ganglia&quot;),ylim=c(0,0.025),ylab=&quot;Prior inclusion&quot;)</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-11-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>y1 &lt;- results_df$pve_Brain_Cerebellum
y2 &lt;- results_df$pve_Brain_Hippocampus
y3 &lt;- results_df$pve_Brain_Caudate_basal_ganglia
y4 &lt;- results_df$pve_Brain_Cerebellar_Hemisphere
y5 &lt;- results_df$pve_Brain_Cortex
y6 &lt;- results_df$pve_Brain_Hypothalamus
y7 &lt;- results_df$pve_Brain_Putamen_basal_ganglia


truth &lt;- rbind(c(1,0.03),c(2,0.03),c(3,0.03),c(4,0),c(5,0),c(6,0),c(7,0))
est &lt;- rbind(cbind(1,y1),cbind(2,y2),cbind(3,y3),cbind(4,y4),cbind(5,y5),cbind(6,y6),cbind(7,y7))
plot_par_7(truth,est,xlabels = c(&quot;Cerebellum&quot;,&quot;Hippocampus&quot;,&quot;Caudate_basal_ganglia&quot;,&quot;Cerebellar_Hemisphere&quot;,&quot;Cortex&quot;,&quot;Hypothalamus&quot;,&quot;Putamen_basal_ganglia&quot;),ylim=c(0,0.06),ylab=&quot;PVE&quot;)</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-12-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>f1 &lt;- plot_PIP(configtag, runtag, paste(1, 1:5, sep = &quot;-&quot;), main = &quot;&quot;)
f1</code></pre>
<p><img src="figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-13-1.png" width="384" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/2f6c0ddf06160fac3acc19edf5993a2cf12b93fa/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-13-1.png" target="_blank">2f6c0dd</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
<tr>
<td>
<a href="https://github.com/sq-96/multigroup_ctwas_analysis/blob/0a3257962b93ddb0326fa6725315cbb69d97fc97/docs/figure/simulation_seven_tissues_correlated.Rmd/unnamed-chunk-13-1.png" target="_blank">0a32579</a>
</td>
<td>
sq-96
</td>
<td>
2023-12-30
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.0 (2021-05-18)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /software/openblas-0.3.13-el7-x86_64/lib/libopenblas_haswellp-r0.3.13.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] plyr_1.8.8        ggpubr_0.6.0      plotrix_3.8-4     cowplot_1.1.1    
[5] ggplot2_3.4.0     data.table_1.14.6 ctwas_0.1.35      workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.9       lattice_0.20-44  tidyr_1.3.0      getPass_0.2-2   
 [5] ps_1.7.2         assertthat_0.2.1 rprojroot_2.0.3  digest_0.6.31   
 [9] foreach_1.5.2    utf8_1.2.2       R6_2.5.1         backports_1.2.1 
[13] evaluate_0.19    highr_0.9        httr_1.4.4       pillar_1.8.1    
[17] rlang_1.1.1      rstudioapi_0.14  car_3.1-1        whisker_0.4.1   
[21] callr_3.7.3      jquerylib_0.1.4  Matrix_1.3-3     rmarkdown_2.19  
[25] labeling_0.4.2   stringr_1.5.0    munsell_0.5.0    broom_1.0.2     
[29] compiler_4.1.0   httpuv_1.6.7     xfun_0.35        pkgconfig_2.0.3 
[33] htmltools_0.5.4  tidyselect_1.2.0 tibble_3.1.8     logging_0.10-108
[37] codetools_0.2-18 fansi_1.0.3      crayon_1.5.2     dplyr_1.0.10    
[41] withr_2.5.0      later_1.3.0      grid_4.1.0       jsonlite_1.8.4  
[45] gtable_0.3.1     lifecycle_1.0.3  DBI_1.1.3        git2r_0.30.1    
[49] magrittr_2.0.3   scales_1.2.1     carData_3.0-4    cli_3.6.1       
[53] stringi_1.7.8    cachem_1.0.6     farver_2.1.0     ggsignif_0.6.4  
[57] fs_1.5.2         promises_1.2.0.1 pgenlibr_0.3.2   bslib_0.4.1     
[61] vctrs_0.6.3      generics_0.1.3   iterators_1.0.14 tools_4.1.0     
[65] glue_1.6.2       purrr_1.0.2      abind_1.4-5      processx_3.8.0  
[69] fastmap_1.1.0    yaml_2.3.6       colorspace_2.0-3 rstatix_0.7.2   
[73] knitr_1.41       sass_0.4.4      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
