---
title: "Region Merge Test"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r setup dir}
library(ctwas)
library(data.table)
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_basic_plots.R")
source("/project/xinhe/shengqian/cTWAS_simulation/summarize_ctwas_plots.R")

outputdir1 = "/project/xinhe/shengqian/cTWAS_simulation/simulation_one_tissue_oldmergeoff/"
outputdir2 = "/project/xinhe/shengqian/cTWAS_simulation/simulation_one_tissue_oldmergeon/"
outputdir3 = "/project/xinhe/shengqian/cTWAS_simulation/simulation_one_tissue_mergeoff/"
runtag = "ukb-s80.45-3"
configtag = 2
```

## PIP calibration
### PIP plot under different settings.
UKBB individual genotype data. PredictedDB Liver weights. UKBB LD matrix. Gene PVE = 0.05 and SNP PVE = 0.3. 


```{r pipplotfunc echo=F}
plot_PIP <- function(outputdir, configtag, runtag,  simutags, ...){
   phenofs <- paste0(outputdir, "ukb-s80.45-3", "_simu", simutags, "-pheno.Rd")
   susieIfs <- paste0(outputdir, runtag, "_simu",simutags, "_config", configtag,"_LDR.susieIrss.txt")

   f1 <- caliPIP_plot(phenofs, susieIfs, ...) 
   return(f1)
}
```

```{r pipplot, fig.width= 12, fig.height= 4, echo=F}
simutaglist = list(paste(1, 1:5, sep = "-"), paste(1, 1:5, sep="-"), paste(1, 1:5, sep="-"))
f1 <- plot_PIP(outputdir1, configtag, runtag, simutaglist[[1]], main = "old merge off")
f2 <-  plot_PIP(outputdir2, configtag, runtag, simutaglist[[2]], main = "merge on")
f3 <-  plot_PIP(outputdir3, configtag, runtag, simutaglist[[3]], main = "new merge off")
gridExtra::grid.arrange(f1, f2, f3, ncol =3)
```